server/core/src/user_validation_utilities.cpp

```diff
@@ -0,0 +1,76 @@
+#include "rodsConnect.h"
+#include "user_validation_utilities.hpp"
+
+#define IRODS_QUERY_ENABLE_SERVER_SIDE_API
+#include "query_builder.hpp"
+
+#include "fmt/format.h"
+
+#include <regex>
+
+namespace irods::user
+{
+    auto get_type(RsComm& _comm, const std::string_view _user_name) -> std::string
+    {
+        char user[NAME_LEN]{};
+        parseUserName(_user_name.data(), user, nullptr);
+
+        const std::string qstr = fmt::format("SELECT USER_TYPE where USER_NAME = '{}'", user);
```

Comment:
same as other comment... what happens if there is both a local and a remote alice?

Reply:
Added zone name to the query to avoid duplicates.